File: Noxa.Emulation.Psp.Bios.NativeHLE/KMemoryBlock.cpp
// ----------------------------------------------------------------------------
// PSP Player Emulation Suite
// Copyright (C) 2006 Ben Vanik (noxa)
// Licensed under the LGPL - see License.txt in the project root for details
// ----------------------------------------------------------------------------

#include "Stdafx.h"
#include "KMemoryBlock.h"
#include <malloc.h>
#include <string.h>

using namespace System;
using namespace Noxa::Emulation::Psp;
using namespace Noxa::Emulation::Psp::Bios;

KMemoryBlock::KMemoryBlock( KPartition* partition, uint address, uint size )
{
	Partition = partition;
	Address = address;
	Size = size;
	UpperBound = address + size;
	IsFree = true;
	Name = NULL;
}

KMemoryBlock::~KMemoryBlock()
{
	SAFEFREE( Name );
}

File: trunk/Noxa.Emulation.Psp.Bios.ManagedHLE/Kernel Types/KMemoryBlock.cs
// ----------------------------------------------------------------------------
// PSP Player Emulation Suite
// Copyright (C) 2006 Ben Vanik (noxa)
// Licensed under the LGPL - see License.txt in the project root for details
// ----------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Text;

namespace Noxa.Emulation.Psp.Bios.ManagedHLE
{
	// NOTE: most memory blocks will not have UIDs!
	
	class KMemoryBlock : KHandle
	{
		public readonly KPartition Partition;
		public string Name;

		public uint Address;
		public uint Size;
		public uint UpperBound
		{
			get
			{
				return Address + Size;
			}
		}

		public bool IsFree;

		public KMemoryBlock( KPartition partition, uint address, uint size, bool isFree )
		{
			Partition = partition;

			Address = address;
			Size = size;

			IsFree = isFree;
		}
	}
}
